### PR TITLE
Merge vm/ test suite with main test suite

### PR DESCRIPTION
**Pull request by howardjohn**
As part of ongoing efforts to minimize test time, we are moving pilot
tests (and soon others) to share a common set of deployments. This
merges the VM tests into this suite. The same tests are ran (traffic
shifting, DNS tests, pod <-> vm, and extra OSes), just in a more
performant manner

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
10 min test run with this PR: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/25754/cache-integ-pilot-k8s-tests_istio/112

File: tests/integration/pilot/vm_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	// Note - bionic is not here as its the default
+	return []string{"app_sidecar_ubuntu_xenial", "app_sidecar_ubuntu_focal",
+		"app_sidecar_debian_9", "app_sidecar_debian_10", "app_sidecar_centos_8"}
+}
+
+func TestVmOSPost(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("traffic.reachability").
+		Label(label.Postsubmit).
```
Comment (stevenctl) [inline tests/integration/pilot/vm_test.go:37]: https://github.com/istio/istio/pull/25432

Individual tests in the pilot/ package should either utilize multiple clusters (see ) or have RequiresSingleCluster until they're converted. 

I need to document this somewhere/make it known. I think an integration testing best practices guide would be good; examples for sharing suite-wide echo instances, making your tests exercise all clusters etc and examples of things that can cause long test times.

Comment (howardjohn) [reply tests/integration/pilot/vm_test.go:37]: Ack, will add. FWIW the existing test I am moving did not have this either

Comment (stevenctl) [reply tests/integration/pilot/vm_test.go:37]: Yeah it's just the root of pilot/ that has multicluster allowed at the suite level.

Comment (howardjohn) [reply tests/integration/pilot/vm_test.go:38]: I see the issue, thanks. I fixed this now

**Pull request merged**